homeassistant/components/tplink/light.py

```diff
@@ -219,6 +230,12 @@ def _get_unique_id(self) -> str:
             # https://github.com/home-assistant/core/blob/2021.9.7/homeassistant/components/tplink/common.py#L86
             return legacy_device_id(device)
 
+        # Newer devices can have child lights. While there isn't currently
+        # an example of a device with more than one light we use the device_id
+        # for consistency and future proofing
```

Comment:
I'm wondering if this would cause duplicates if such a device would suddenly get released?

Reply:
The `device_id` is unique to the parent and the child devices for both `smart` and `iot` so this should be ok.